doc/design/block.md

```diff
@@ -0,0 +1,128 @@
+# Block design
+
+In computer programming, a block is a lexical structure of source code which is grouped together.
```

Comment:
According to [Wikipedia](https://en.wikipedia.org/wiki/Block_(programming))

> a block or code block is a lexical structure of source code which is grouped together.  

In C++ and Java, a block is represented by a pair of curly braces, and is used in control structures like if, for, and function definitions.

Blocks do not only group source code, but also

> narrow the lexical scope of variables so that they do not conflict with variables having the same name used elsewhere in a program.

Reply:
yes, this Block will be used by `if_else_op`, `while_op` and `RNNOp`, `RNNOp` will use a `Block` as a function.

Reply:
I wrote this segment for your reference how to change the design doc. I was trying to rewrite this paragraph.